### ISSUE TITLE
variable-definitions in electric-imp.settings should allow all data types

### ISSUE THREAD

**Issue by thomaskoehn-kisi** [electricimp/Builder#48]
It seems to only work for strings right now. I cannot specify other Builder data types like booleans, arrays, or tables.

```
{ "builder-settings": { ...,
                        "variable-definitions": { "key1": { "inner_key1" : [true, 25.3] } },
                        ... }
}
```

**Comment by ppetrosh** [comment]
There are no types in Builder language. It's a text preprocessor and works with strings only by design.

**Comment by ppetrosh** [comment]
In the string value you can specify whatever you want to inject, be it boolean, table or array code.

**Comment by thomaskoehn-kisi** [comment]
hmm it's just, if I want it to be a boolean and check for it afterwards with an `@if`, it will probably still equate `"false"` to `true`, since the string is defined, right?

**Comment by thomaskoehn-kisi** [comment]
Plus, why this then: https://github.com/electricimp/Builder#types

**Comment by ppetrosh** [comment]
My bad. You are right, the constants can further be used in expressions and can be of types other than string. Even though arrays/maps are not supported the other types should work. I'm sorry, will fix that.

**Comment by ppetrosh** [comment]
Arrays and maps are not supported by Builder. But all other types should now be supported. See https://github.com/electricimp/ElectricImp-Sublime/pull/56

Example of the Squirrel code:

```
local val = "var_value";
// as var name
server.log(@{str}); // "var_value"
// string var
server.log("@{str}"); // "val"
// number vars
server.log(@{num1}); // 34
server.log(@{num2}); // 34.356
server.log(@{num3}); // 3E4 -> 30000
server.log(@{num4}); // 3e-2 -> 0.03
// bool vars
server.log(@{bt}); // true
server.log(@{bf}); // false
// null value
server.log(@{numn}); //null
```

Example of the configuration:

```
        "variable-definitions": {
            "num4": 0.03, 
            "num3": 30000.0, 
            "num2": 34.456, 
            "num1": 34, 
            "numn": null, 
            "bt": true, 
            "str": "val", 
            "bf": false
        }
```

**Comment by ppetrosh** [comment]
fixed in v2.1.0
